crates/astria-conductor/src/block_verifier.rs

```diff
@@ -0,0 +1,616 @@
+            .height
+            .parse()
+            .wrap_err("failed to parse header height")?;
+        let validator_set = self
+            .tendermint_client
+            .get_validator_set(height - 1)
+            .await
+            .wrap_err("failed to get validator set")?;
+
+        // find proposer address for this height
+        let expected_proposer_address = validator_set
+            .get_proposer()
+            .wrap_err("failed to get proposer from validator set")?
+            .address;
+
+        // verify the namespace data signing public key matches the proposer address
+        let res_address = public_key_to_bech32_address(&data.public_key.0)
+            .wrap_err("failed to convert namespace data public key to address")?;
+        ensure!(
+            res_address == expected_proposer_address,
+            format!(
+                "public key mismatch: expected {}, got {}",
+                expected_proposer_address, res_address
+            ),
+        );
+
+        Ok(())
+    }
+
+    /// performs various validation checks on the SequencerBlock received from either gossip or
+    /// Celestia.
+    ///
+    /// checks performed:
+    /// - the proposer of the sequencer block matches the expected proposer for the block height
+    ///   from tendermint
+    /// - the signer of the SignedNamespaceData the proposer
+    /// - the signature is valid
+    /// - the root of the markle tree of all the header fields matches the block's block_hash
+    /// - the root of the merkle tree of all transactions in the block matches the block's data_hash
+    /// - validate the block was actually finalized; ie >2/3 stake signed off on it
+    pub(crate) async fn validate_sequencer_block(
+        &self,
+        block: &SequencerBlock,
+    ) -> eyre::Result<()> {
+        // sequencer block's height
+        let height: u64 = block
+            .header
+            .height
+            .parse()
+            .wrap_err("failed to parse height")?;
+
+        // get validator set for this height
+        let validator_set = self
+            .tendermint_client
+            .get_validator_set(height - 1)
+            .await
+            .wrap_err("failed to get validator set")?;
+
+        // find proposer address for this height
+        let expected_proposer_address = validator_set
+            .get_proposer()
+            .wrap_err("failed to get proposer from validator set")?
+            .address;
+
+        // check if the proposer address matches the sequencer block's proposer
+        let received_proposer_address = bech32::encode(
+            METRO_VALIDATOR_ADDRESS_PREFIX,
+            block.header.proposer_address.0.to_base32(),
+            Variant::Bech32,
+        )
+        .wrap_err("failed converting bytes to bech32 address")?;
+
+        ensure!(
+            received_proposer_address == expected_proposer_address,
+            format!(
+                "proposer address mismatch: expected {}, got {}",
```

Comment:
As above - I think you don't need to explicitly format here.

But even if you wanted to, you should probably prefer this:
```rust
        ensure!(
            received_proposer_address == expected_proposer_address,
            "proposer address mismatch: expected `{expected_propser_address}`, got `{received_proposer_address}`",
    )